src/libc/string.rs

```diff
@@ -148,20 +148,16 @@ fn strstr(env: &mut Environment, string: ConstPtr<u8>, substring: ConstPtr<u8>)
             return Ptr::null(); // NULL
         }
 
-        match char_string.cmp(&char_substring) {
-            Ordering::Less => {}
-            Ordering::Greater => {}
-            Ordering::Equal => {
-                let char_substring_plusone = env.mem.read(substring + 1);
-                let char_string_plusone = env.mem.read(string + offset + 1);
-
-                loop {
-                    if char_substring == 0 {
-                        return string + offset;
-                    }
-                    if char_string_plusone != char_substring_plusone {
-                        break;
-                    }
+        if char_string == char_substring {
+            let char_substring_plusone = env.mem.read(substring + 1);
+            let char_string_plusone = env.mem.read(string + offset + 1);
+
+            loop {
```

Comment:
This doesn't look right, you're not incrementing at all within the loop…

Reply:
Whoops! Should be fixed now.